cpp/src/arrow/compute/kernels/scalar_temporal.cc

```diff
@@ -466,8 +492,90 @@ struct Nanosecond {
 // Convert timestamps to a string representation with an arbitrary format
 
 #ifndef _WIN32
-template <typename Duration>
+template <typename Duration, typename InType>
 struct Strftime {
```

Comment:
nit: since this is for dates only, maybe have the base `struct Strftime` have no implementation, and implement both dates and timestamps as specializations?

Reply:
Actually, given that the regular strftime kernel won't format timestamps without timezones, I wonder if it's meaningful to have strftime work on dates. I would say if you want a string representation of a date, you should cast, since strftime would let you do misleading things like trying to extract the hour from a date.

Reply:
I think you are right. Do you know if we have tests for casting date32/64 to strings?

Reply:
We do, they're a bit thin though: https://github.com/apache/arrow/blob/5ead37593472c42f61c76396dde7dcb8954bde70/cpp/src/arrow/compute/kernels/scalar_cast_test.cc#L1245

Reply:
> Actually, given that the regular strftime kernel won't format timestamps without timezones, I wonder if it's meaningful to have strftime work on dates.

There's work to have Strftime work on timestamps without timezones (assuming UTC)
https://github.com/apache/arrow/pull/10998. It's a philosophical discussion if it's correct or not but it will be available.

Reply:
Hmm. I take it back then. Python actually supports this too: https://docs.python.org/3/library/datetime.html#datetime.date.strftime

Reply:
Sorry for the back-and-forth @aucahuasi. However, it might be prudent to split strftime support into a separate JIRA to minimize conflicts with the existing PR and so we can evaluate how best to share the implementation (to avoid too much code duplication).

Reply:
No worries @lidavidm :)
Indeed, it makes sense to have a new jira. I can create the ticket if there is none.

Reply:
I don't think there's a JIRA so please do file one.

Reply:
Here is the jira ticket for strftime, thanks @lidavidm @rok 
https://issues.apache.org/jira/browse/ARROW-13916

Reply:
The Strftime ticket was merged yesterday so maybe something has changed here. Dunno.